src/components/exchange/ExchangeNativeField.tsx

```diff
@@ -1,30 +1,63 @@
-import React, { useCallback, useMemo, useState } from 'react';
-import { TouchableWithoutFeedback } from 'react-native';
+import React, {
+  ForwardRefRenderFunction,
+  MutableRefObject,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { TextInput, TouchableWithoutFeedback } from 'react-native';
 import { Row } from '../layout';
 import { Text } from '../text';
 import ExchangeInput from './ExchangeInput';
 import { useColorForAsset } from '@rainbow-me/hooks';
 import { supportedNativeCurrencies } from '@rainbow-me/references';
 import styled from '@rainbow-me/styled-components';
 import { fonts } from '@rainbow-me/styles';
+import { useTheme } from '@rainbow-me/theme';
 
-const CurrencySymbol = styled(Text).attrs(({ height, color }) => ({
-  color: color,
-  letterSpacing: 'roundedTight',
-  lineHeight: height,
-  size: 'larger',
-  weight: 'regular',
-}))(android ? { marginBottom: 1.5 } : {});
+interface CurrencySymbolProps {
+  height: number;
+  color: string;
+}
+
+const CurrencySymbol = styled(Text).attrs(
+  ({ height, color }: CurrencySymbolProps) => ({
+    color: color,
+    letterSpacing: 'roundedTight',
+    lineHeight: height,
+    size: 'larger',
+    weight: 'regular',
+  })
+)(android ? { marginBottom: 1.5 } : {});
 
 const NativeInput = styled(ExchangeInput).attrs({
   letterSpacing: fonts.letterSpacing.roundedTight,
   size: fonts.size.larger,
   weight: fonts.weight.regular,
 })({
-  height: ({ height }) => height,
+  height: ({ height }: { height: number }) => height,
 });
 
-const ExchangeNativeField = (
+interface ExchangeNativeFieldProps {
+  address: string;
+  editable: boolean;
+  height: number;
+  loading: boolean;
+  nativeAmount: string | null;
+  nativeCurrency: string;
+  onFocus: ({ target }: { target: Element }) => void;
+  setNativeAmount: (value: string | null) => void;
+  updateOnFocus: boolean;
+  testID: string;
+}
+
+type SupportedNativeCurrencies = keyof typeof supportedNativeCurrencies;
```

Comment:
may be worth moving this to somewhere in `entities` and importing

Reply:
Found a similar type that was created already so swapped to using that now